subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/ForkingTestClassProcessor.java

```diff
@@ -110,6 +110,8 @@ RemoteTestClassProcessor forkProcess() {
             moduleRegistry.getModule("gradle-native").getImplementationClasspath().getAsURLs(),
             moduleRegistry.getModule("gradle-testing-base").getImplementationClasspath().getAsURLs(),
             moduleRegistry.getModule("gradle-testing-jvm").getImplementationClasspath().getAsURLs(),
+            moduleRegistry.getModule("gradle-testing-junit-platform").getImplementationClasspath().getAsURLs(),
+            moduleRegistry.getExternalModule("junit-platform-launcher").getImplementationClasspath().getAsURLs(),
```

Comment:
does the inclusion of a Java 8-compiled dependency present any problems if the worker JVM is a Java 7 one?

Reply:
Good catch, that's what I'm going to do. Main process won't complain about this issue because it will pass the Java8-jar url to test worker and won't load this jar itself. However, if test worker process is Java 7, when it tries to load JUnit 5 classes , errors will happen. We need to add some warnings which tells user, "you're trying to use JUnit 5 on Java 7, we can't allow you to do that".

Reply:
The Java8 classes are instantiated via reflection here: https://github.com/gradle/gradle/blob/cd73ee4cb0e7074a6a620b43e8b059cbc11fde43/subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestFramework.java#L95

And I've already added [some logic](https://github.com/gradle/gradle/blob/cd73ee4cb0e7074a6a620b43e8b059cbc11fde43/subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestFramework.java#L53) to forbid users to start JUnit5 in Java 7:

```
        if (!JavaVersion.current().isJava8Compatible()) {
            throw new UnsupportedJavaRuntimeException("Running JUnit platform requires Java 8+, please configure your test java executable with Java 8 or higher.");
        }
```